test/test_framework.py

```diff
@@ -718,6 +729,13 @@ def test_helper_properties(self):
         self.assertEqual(my_obj.meta, framework.meta)
         self.assertEqual(my_obj.charm_dir, framework.charm_dir)
 
+    def test_ban_concurrent_frameworks(self):
+        framework = self.create_framework()
+        with self.assertRaises(Exception) as cm:
+            framework_copy = self.create_framework()
+            self.fail(f'frameworks {framework} and {framework_copy} got instantiated successfully')
+        self.assertTrue('database is locked' in str(cm.exception))
```

Comment:
This is better, as when it fails it will tell you what actually was in the string instead:
```
self.assertIn('database is locked', str(cm.exception))
```
But in this case, it looks like it might be simply:
```
self.assertEqual('database is locked', str(cm.exception))
```

Reply:
https://github.com/canonical/operator/pull/141/commits/62c919c9f9687de0b4cd3391bcbef482a511d9a5